src/main.cpp

```diff
@@ -3765,6 +3770,32 @@ std::string GetWarnings(const std::string& strFor)
 // Messages
 //
 
+static std::map<std::string, size_t> maxMessageSizes = boost::assign::map_list_of
+    ("addr", 2+1000*30)
+    ("getaddr",0)
+    ("getdata", 2+MAX_INV_SZ*36)
+    ("inv", 2+MAX_INV_SZ*36)
+    ("mempool",0)
+    ("notfound", 2+MAX_INV_SZ*36)
+    ("ping",8)
+    ("pong",8)
+    ("verack", 0)
+    ;
+
+bool static SanityCheckMessage(CNode* peer, const CNetMessage& msg)
```

Comment:
Why did you choose to not directly pass the NodeId? It seems it's the only field used in here.